### PR TITLE
Prevent quantization on appendable

### PR DESCRIPTION
**Pull request by generall**

### CODE REVIEW

File: lib/segment/src/segment_constructor/segment_constructor_base.rs [hunk 1]
```diff
@@ -108,7 +108,8 @@ fn create_segment(
             )?,
         };
 
-        if config.quantization_config(vector_name).is_some() {
+        if config.quantization_config(vector_name).is_some() && config.quantization_config.is_some()
```
Comment (timvisee) [inline lib/segment/src/segment_constructor/segment_constructor_base.rs:111]: I don't see how this change would prevent quantization on appendable segments. Could you elaborate?

Note that the `config.quantization_config(vector_name)` function returns the vector specific config, and if it doesn't exist, the global config.

Comment (generall) [reply lib/segment/src/segment_constructor/segment_constructor_base.rs:111]: So when we were designing configuration for individual named vectors, we had in mind the following scenario:

configuration of individual vectors( if exists ) should be preferred over global configuration (if global configuration enabled)

See how indexing works: it only applies named configuration if there is a global one. But for quantization we broke this logic and tried to always load quantized data no matter if global config is set or not.

This new condition fixes his logic: we only load quantization if global config is set.

Worth mentioning that currently, if there quantization is configured in collection - it is always set in named config. 
So currently - each segment always tries to load quntization.

**Comment by generall** [comment]
This PR is temporary fix meant to resolve https://github.com/qdrant/qdrant/issues/1831

In the next minor release `SegmentConfig` should be upgraded to better version (with migration)

File: lib/segment/src/segment_constructor/segment_constructor_base.rs (hunk 1 above)
Comment (timvisee) [reply lib/segment/src/segment_constructor/segment_constructor_base.rs:111]: Thanks for the explanation.

Then, shouldn't we change `config.quantization_config(vector_name)` to only return a configuration when a global one is set as well? Then we could keep this line intact, and possibly prevent confusion in the future.

Comment (generall) [reply lib/segment/src/segment_constructor/segment_constructor_base.rs:111]: maybe, but I want to release the fix Asap, to fix the problem for users. refactoring can be done in minor release

Comment (timvisee) [reply lib/segment/src/segment_constructor/segment_constructor_base.rs:111]: Okay! :smiley:

**Review by timvisee — APPROVED** [review]

**Review by IvanPleshkov — APPROVED** [review]

**Pull request merged**